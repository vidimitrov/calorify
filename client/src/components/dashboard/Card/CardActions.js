import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import IconButton from '@material-ui/core/IconButton';
import Edit from '@material-ui/icons/Edit';
import Delete from '@material-ui/icons/Delete';

const Wrapper = styled.div`
  width: 100px;

  ${props => props.wide && css`
    width: 150px;
  `}

  @media (max-width: 420px) {
    width: 80px;
  }
`;

function CardActions({
  onEditHandler,
  onDeleteHandler,
  customAction,
}) {
  return (
    <Wrapper wide={!!customAction}>
      {customAction
        && (
          <IconButton
            name={customAction.name ? customAction.name : ''}
            title={customAction.title ? customAction.title : ''}
            onClick={customAction.handler}
            color="default"
          >
            {customAction.icon}
          </IconButton>
        )
      }
      <IconButton
        onClick={onEditHandler}
        name="edit-meal"
        color="default"
      >
        <Edit />
      </IconButton>
      <IconButton
        onClick={onDeleteHandler}
        name="delete-meal"
        color="default"
      >
        <Delete />
      </IconButton>
    </Wrapper>
  );
}

CardActions.propTypes = {
  onEditHandler: PropTypes.func.isRequired,
  onDeleteHandler: PropTypes.func.isRequired,
  customAction: PropTypes.shape({
    icon: PropTypes.element.isRequired,
    handler: PropTypes.func.isRequired,
    title: PropTypes.string,
  }),
};

CardActions.defaultProps = {
  customAction: null,
};

export default CardActions;
