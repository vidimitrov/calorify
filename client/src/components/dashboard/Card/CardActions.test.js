import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import CardActions from './CardActions';

it('Dashboard/Card/CardActions component renders correctly', () => {
  const tree = renderer.create(
    <CardActions onEditHandler={() => { }} onDeleteHandler={() => { }} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
