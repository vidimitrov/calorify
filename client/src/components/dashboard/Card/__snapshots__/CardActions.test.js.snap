// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard/Card/CardActions component renders correctly 1`] = `
.c0 {
  width: 100px;
}

<div
  className="c0"
>
  <button
    className="MuiButtonBase-root-7 MuiIconButton-root-1"
    disabled={false}
    name="edit-meal"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiIconButton-label-6"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-10"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-19"
    />
  </button>
  <button
    className="MuiButtonBase-root-7 MuiIconButton-root-1"
    disabled={false}
    name="delete-meal"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiIconButton-label-6"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-10"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-19"
    />
  </button>
</div>
`;
